### PR TITLE
Expose source and destination directory variables to pre and post build scripts

### PR DESCRIPTION
**Pull request by kichalla**

### CODE REVIEW

File: src/BuildScriptGenerator/Node/NodeBashBuildSnippet.sh.tpl [hunk 1]
```diff
@@ -107,24 +107,28 @@ then
...
-		START_TIME=$SECONDS
-		# Make the contents of the node_modules folder appear in the zip file, not the folder itself
-		cd node_modules
-		{{ CompressNodeModulesCommand }} ../$zippedModulesFileName .
-		ELAPSED_TIME=$(($SECONDS - $START_TIME))
-		echo "Done in $ELAPSED_TIME sec(s)."
+		if [ -f $zippedModulesFileName ]; then
+			echo
+			echo "File '$zippedModulesFileName' already exists under '$SOURCE_DIR'. Deleting it ..."
+			rm -f $zippedModulesFileName
```
Comment (gpcastro) [inline src/BuildScriptGenerator/Node/NodeBashBuildSnippet.sh.tpl]: I think that the zip tool optimizes for this case when the file is already there, only adding files.

File: tests/Oryx.BuildImage.Tests/PythonSampleAppsTest.cs [hunk 2]
```diff
@@ -859,6 +859,71 @@ public void Build_ExecutesPreAndPostBuildScripts_UsingEnvironmentVariables()
                 result.GetDebugInfo());
         }
 
+        [Fact]
```
Comment (arroyc) [inline tests/Oryx.BuildImage.Tests/PythonSampleAppsTest.cs]: which version of python does this test target

File: src/BuildScriptGenerator/Node/NodeBashBuildSnippet.sh.tpl (hunk 1 above)
Comment (kichalla) [reply src/BuildScriptGenerator/Node/NodeBashBuildSnippet.sh.tpl]: I am not doing this change in this PR any more and only exposing the source and destination directories.

File: tests/Oryx.BuildImage.Tests/PythonSampleAppsTest.cs (hunk 2 above)
Comment (kichalla) [reply tests/Oryx.BuildImage.Tests/PythonSampleAppsTest.cs]: I think we already discussed about this offline.

**Pull request merged**